runelite-client/src/main/java/net/runelite/client/plugins/twitch/TwitchPlugin.java

```diff
@@ -0,0 +1,419 @@
+				}
+				//If chat is opaque and there are no room states
+				else if (client.getGameState().equals(GameState.LOGGED_IN))
+				{
+					client.addChatMessage(ChatMessageType.TRADE, "",
+							"<col=000000>[</col><col=0000FF>Twitch</col>] " + badges + config.channelName() + ": <col=" + Integer.toHexString(config.chatOColor().getRGB() & 0xFFFFFF) + ">" + message + "</col>", "Twitch");
+				}
+			}
+		}
+	}
+
+	//Shut off current running irc client
+	public void twitchShutdown(Client c)
+	{
+		c.shutdown();
+	}
+
+	//Triggers when a message is sent to twitch chat
+	@Handler
+	public void onMessage(ChannelMessageEvent a)
+	{
+		String username = a.getActor().getUserString();
+
+		//Sub length badges
+		if (a.getOriginalMessages().get(0).getTag("badges").toString().contains("subscriber/0"))
+		{
+			username = "<img=9>" + username;
+		}
+		else if (a.getOriginalMessages().get(0).getTag("badges").toString().contains("subscriber/3"))
+		{
+			username = "<img=8>" + username;
+		}
+		else if (a.getOriginalMessages().get(0).getTag("badges").toString().contains("subscriber/6"))
+		{
+			username = "<img=7>" + username;
+		}
+		else if (a.getOriginalMessages().get(0).getTag("badges").toString().contains("subscriber/12"))
+		{
+			username = "<img=6>" + username;
+		}
+		else if (a.getOriginalMessages().get(0).getTag("badges").toString().contains("subscriber/24"))
+		{
+			username = "<img=5>" + username;
+		}
+		else if (a.getOriginalMessages().get(0).getTag("badges").toString().contains("subscriber/48"))
+		{
+			username = "<img=4>" + username;
+		}
+
+		//Moderator badge
+		if (a.getOriginalMessages().get(0).getTag("badges").toString().contains("moderator/1"))
+		{
+			username = "<img=26>" + username;
+		}
+
+		//Streamer badge
+		if (a.getOriginalMessages().get(0).getTag("badges").toString().contains("broadcaster/1"))
+		{
+			username = "<img=27>" + username;
+		}
+
+		//Show the message in game with color based on if the chat is transparent or not
+		String message = a.getMessage();
+		if (isChatboxTransparent() && client.getGameState().equals(GameState.LOGGED_IN))
+		{
+			client.addChatMessage(ChatMessageType.TRADE, "",
+					"<col=ffffff>[</col><col=9070ff>Twitch</col>] " + username + ": <col=" + Integer.toHexString(config.chatTColor().getRGB() & 0xFFFFFF) + ">" + message + "</col>", "Twitch");
+		}
+		else if (client.getGameState().equals(GameState.LOGGED_IN))
+		{
+			client.addChatMessage(ChatMessageType.TRADE, "",
+					"<col=000000>[</col><col=0000FF>Twitch</col>] " + username + ": <col=" + Integer.toHexString(config.chatOColor().getRGB() & 0xFFFFFF) + ">" + message + "</col>", "Twitch");
+		}
+	}
+
+	//User and room information update events
+	//Triggers whenever the user's state is updated or they join a room
+	@Handler
+	public void onUserStateEvent(UserStateEvent u)
```

Comment:
Is this handler necessary? Seems like you could handle all of this in onMessage

Reply:
Like it says in the comment if they swap channels it won't update until they send a message so if they swap from a channel they're a sub in to another that's in sub mode that they're not a sub in it'll show the first message they send in chat when it's not actually being sent. If that makes any sense at all